### ISSUE TITLE
RQ2の予測モデル構築前処理としてデータバランシング適用

### ISSUE THREAD

**Issue by y629** [posl/DNN-repair-break-prediction#17]
SMOTE
Under sampling

**Comment by y629** [comment]
既存コードで行われていたsampling (repairs, breaks共通)
- 多数派クラスの数を *(rus_ratio) にするようにランダムアンダーサンプリング．rus_ratio = データセットのサイズ (DNN構築時のfolds数) に依存するが，0.2 or 0.4 (適当)

追加したいsampling
- アンダーサンプリングの後にSMOTEでオーバーサンプリングする．SMOTEの原著論文でもアンダーサンプリングと組み合わせることが有効であると示唆されている．少数派クラスに有利な予測結果になるか？
- 結果見た感じ，training setに対してはよくなりそうだけど test set に対してはそんなに変わらない (over fitting してるだけ?)

**Comment by y629** [comment]
Discussionのサブセクションadditional experimentに追加する

**Comment by y629** [comment]
RQ2ではRUSだけ
additional analysisでは
- RUS + SMOTE
- なにもしない
を追加したい

**Comment by y629** [comment]
何もしない場合も追加した．

明日結果をどうまとめるか考えて論文に反映．
